Update README.md

liannmfinlayson-arch opened the pull request:
Add what rebase is

liannmfinlayson-arch committed:
Update README.md

liannfinlayson26 commented on README.md:
```diff
@@ -1,3 +1,4 @@
 # practice-rebase-demo
 
-First commit
+What is rebase?
```
Put this title in ## to be more styled

liannfinlayson26 commented on README.md:
```diff
@@ -1,3 +1,4 @@
 
-First commit
+What is rebase?
+Rebase puts the latest commit on top of the feature/main commit. To avoid three way merge.
```
Provide examples 
Define what merge is and also three way merge. Provide examples as well

liannmfinlayson-arch committed:
Update README.md

Added definition and examples of:
1. Merge: Fast-forward merge and three-way merge
2. Rebase

liannmfinlayson-arch commented:
Resolve all comments. Could you take another look

liannfinlayson26 commented on README.md:
```diff
@@ -1,3 +1,58 @@
+git merge feature
+
+History before merge:
+A---B---C (main)
```
```suggestion
## Example: Three-way merge
#### main has commits A-B-C
#### feature branched off at B and added D and E

git checkout main
git merge feature

History before merge:
A---B---C (main)
```

liannmfinlayson-arch committed:
Update README.md

Change from ### to #### in readme file

Co-authored-by: Liann Finlayson <liannmfinlayson@hotmail.com>
